Source/WebCore/dom/Document.cpp

```diff
@@ -5126,33 +5126,22 @@ void Document::updateViewportUnitsOnResize()
 
 void Document::setNeedsDOMWindowResizeEvent()
 {
-#if ENABLE(FULLSCREEN_API)
-    if (CheckedPtr fullscreenManager = fullscreenManagerIfExists(); fullscreenManager && fullscreenManager->isAnimatingFullscreen()) {
-        fullscreenManager->addPendingScheduledResize(FullscreenManager::ResizeType::DOMWindow);
-        return;
-    }
-#endif
-
     m_needsDOMWindowResizeEvent = true;
     scheduleRenderingUpdate(RenderingUpdateStep::Resize);
 }
 
 void Document::setNeedsVisualViewportResize()
 {
-#if ENABLE(FULLSCREEN_API)
-    if (CheckedPtr fullscreenManager = fullscreenManagerIfExists(); fullscreenManager && fullscreenManager->isAnimatingFullscreen()) {
-        fullscreenManager->addPendingScheduledResize(FullscreenManager::ResizeType::VisualViewport);
-        return;
-    }
-#endif
-
     m_needsVisualViewportResizeEvent = true;
     scheduleRenderingUpdate(RenderingUpdateStep::Resize);
 }
 
 // https://drafts.csswg.org/cssom-view/#run-the-resize-steps
 void Document::runResizeSteps()
 {
+    if (auto page = this->page(); page && page->shouldDeferResizeEvents())
```

Comment:
```suggestion
    if (RefPtr page = this->page(); page && page->shouldDeferResizeEvents())
```

Reply:
Per the Safer C++ rules, you don't have to hold a RefPtr on an object if the function being called is "trivial", defined as "the function being called is inlined". Since these accessors are just inlined methods returning a bool ivar, a ref isn't necessary.

Reply:
I'm more concerned about consistency in this file, than strictly making the static analyzer happy.
There's not a single call to `Document::page()` in Document.cpp that don't use the semantics `RefPtr page = page()` 
Document.cpp has been over several iterations to replace all `auto` with `RefPtr` seems to be a step backward to re-introduce the use of auto. 

If not using RefPtr, the other style use here is more something like `if (page() && page()->method()`

Reply:
That would become then

```suggestion
    if (page() && page()->shouldDeferResizeEvents())
```

Reply:
I'm not going to adopt this style in the patch. It's needlessly wasteful to call `page()` twice just to achieve "consistency". And the Safer Cpp document explicitly allows for this style of call.